reader/views.py

```diff
@@ -3560,6 +3562,43 @@ def profile_follow_api(request, ftype, slug):
     return jsonResponse({"error": "Unsupported HTTP method."})
 
 
+@catch_error_as_json
+def topic_delete_photo(request, file):
+    if not request.user.is_authenticated:
+        return jsonResponse({"error": _("You must be logged in to update a topic photo.")})
+    bucket_name = GoogleStorageManager.TOPICS_BUCKET
+    if file:
+        file = f"topics/{file.split('/')[-1]}"
+    if request.method == "DELETE":
+        if file is None:
+            return jsonResponse({"error": "You cannot remove an image as you haven't selected one yet."})
+        GoogleStorageManager.delete_filename(file, bucket_name)
+        return jsonResponse({"success": "You have successfully removed the image."})
+
+@catch_error_as_json
+def topic_upload_photo(request):
+    from io import BytesIO
+    import uuid
+    import base64
+    if not request.user.is_authenticated:
+        return jsonResponse({"error": _("You must be logged in to update a topic photo.")})
+    bucket_name = GoogleStorageManager.TOPICS_BUCKET
+    file = request.POST.get('file')
+    old_filename = request.POST.get('old_filename')  # delete file from google storage if there is one there
+    if old_filename:
+        old_filename = f"topics/{old_filename.split('/')[-1]}"
+    if request.method == "DELETE":
```

Comment:
What is this DELETE doing versus the separate api endpoint?

Reply:
It seems, from the client code below, that this api view is never called with the DELETE http verb. 
Therefore it doesnt make sense to be checking for it here in the view. 

That being said, the functionality inside `topic_delete_photo()` _should_ be placed in this same API view as a separate path for when the api is called with DELETE - which is when the user wants to completely delete a topic photo, not when they want to replace it during editing - that part is implicitly handled in the code for the upload already.